src/api/c++/z3++.h

```diff
@@ -743,12 +818,10 @@ namespace z3 {
        \brief A Z3 expression is used to represent formulas and terms. For Z3, a formula is any expression of sort Boolean.
        Every expression has a sort.
     */
-    class expr : public ast {
+    class expr final : public ast {
```

Comment:
could a user of the library complain that they can't extend this?

Reply:
Indeed. I could make this either a build option or remove it. It is an optimization _if_ `expr` is not going to be extended. Would you prefer me remove it or make a macro called something like `MAYBE_FINAL` that the user can define before including this file?